spec/_attachments/ic.did

```diff
@@ -182,6 +182,10 @@ type install_code_args = record {
         reinstall;
         upgrade : opt record {
             skip_pre_upgrade : opt bool;
+            wasm_memory_persistence : opt variant {
+                keep;
+                drop;
+            };
```

Comment:
Good idea to have a variant. Then we can add future options (eg. to handle multiple memories selectively)

Reply:
Would it make sense to add type
```
type Mode = variant {...}
```
and use it twice, rather than duplicating. Or maybe that breaks existing (Rust) clients.

Reply:
Maybe we could even sub-summarize the entire upgrade options as an intermediate type:

```
type CanisterUpgradeOptions = record {
     skip_pre_upgrade : opt bool;
     wasm_memory_persistence : opt variant {
        keep;
        drop;
     };
};
```

Reply:
The duplication is due to having install_code and install_chunked_code: I plan to address it separately.

Reply:
https://github.com/dfinity/interface-spec/pull/282